pynicotine/gtkgui/userbrowse.py

```diff
@@ -688,34 +711,338 @@ def on_select_dir(self, selection):
 
         self.set_directory(iterator.user_data)
 
-    def on_folder_key_press_event(self, *args):
+    def on_select_file(self, selection):
+
+        if selection is None:
+            return
+
+        self.num_selected_files = selection.count_selected_rows()
+
+    def on_key_press_event_folder(self, *args):
 
         keyval, keycode, state, widget = get_key_press_event_args(*args)
-        keycodes, mods = parse_accelerator("Right")
+        path, _focus_column = self.FolderTreeView.get_cursor()
 
-        if keycode in keycodes:
-            path, _focus_column = self.FolderTreeView.get_cursor()
-            self.FolderTreeView.expand_row(path, False)
-            return True
+        _keys, mods_primary = parse_accelerator("<Primary>")
+        _keys, mods_shift = parse_accelerator("<Shift>")
+        _keys, mods_alt = parse_accelerator("<Alt>")
+
+        prs_none = prs_ctrl = prs_shift = prs_alt = False
+
+        if state & mods_primary:  # <GDK_CONTROL_MASK>
+            prs_ctrl = True
+
+        if state & mods_shift:  # <GDK_SHIFT_MASK>
+            prs_shift = True
+
+        if state & mods_alt:  # <GDK_MOD1_MASK>
+            prs_alt = True
+
+        if prs_ctrl == prs_shift == prs_alt is False:
+            prs_none = True  # <GDK_MOD2_MASK> = plain
 
+        """ Key Bindings (FolderTreeView) """
+
+        # Left: Collapse folder
         keycodes, mods = parse_accelerator("Left")
 
-        if keycode in keycodes:
-            path, _focus_column = self.FolderTreeView.get_cursor()
-            self.FolderTreeView.collapse_row(path)
+        if keycode in keycodes and (not prs_ctrl) and (not prs_alt):
+            if path is not None and (prs_none):
+                self.FolderTreeView.collapse_row(path)
+                return True
+
+        # Right: Expand folder
+        keycodes, mods = parse_accelerator("Right")
+
+        if (keycode in keycodes and not prs_ctrl and not prs_alt) or (
+            keycode == 21 and prs_none) or (  # EN-US/GB laptop non-shifted +"=" key
+            keycode == 94 and prs_none        # EN-GB "\" key (KeyVal here instead?)
```

Comment:
The "=" has been changed to a keycode.
The "\" might be better off as a KeyVal, because it appears in different places on international keyboards, I'm not sure.

Reply:
I found the answer to this (21="equal"; 94="backslash"; 22="BackSpace"), link:
https://anzeljg.github.io/rin2/book2/2405/docs/tkinter/key-names.html
https://www.tcl.tk/man/tcl8.4/TkCmd/keysyms.html